distage/distage-testkit-scalatest/src/main/scala-2/izumi/distage/testkit/scalatest/AssertZIO.scala

```diff
@@ -28,6 +46,43 @@ object AssertZIO extends AssertZIO {
       import c.universe._
       c.Expr[IO[Nothing, Assertion]](q"_root_.zio.ZIO.succeed(${DistageAssertionsMacro.assert(c)(arg)(prettifier, pos)})($zioTrace)")
     }
+
+    def shortImpl1[T: c.WeakTypeTag](
+      c: blackbox.Context
+    )(effect: c.Expr[IO[Nothing, T]]
+    )(predicate: c.Expr[T => Boolean]
+    )(prettifier: c.Expr[Prettifier],
+      pos: c.Expr[Position],
+      zioTrace: c.Expr[zio.Trace],
+    ): c.Expr[IO[Nothing, Assertion]] = {
+      import c.universe._
+
+      val resultName = TermName(c.freshName("result"))
+      val predicateBody = ReflectionUtil.betaReduceLambda1[T, Boolean](c)(predicate, resultName.toString)
+
+      c.Expr[IO[Nothing, Assertion]](q"$effect.flatMap { ($resultName: ${weakTypeOf[T]}) => assertIO($predicateBody)($prettifier, $pos, $zioTrace) }")
+    }
+
+    def shortImpl2[A: c.WeakTypeTag, B: c.WeakTypeTag](
+      c: blackbox.Context
+    )(effectA: c.Expr[IO[Nothing, A]],
```

Comment:
Could you please make IO2 & ZIO asserts polymorphic by error type (and R for ZIO)

```suggestion
    )(effectA: c.Expr[IO[E, A]],
```